Storage/ThirdParty/SwiftData.swift

```diff
@@ -113,39 +113,33 @@ public class SwiftData {
             conn = ConcreteSQLiteDBConnection(filename: filename, flags: flags.toSQL(), key: self.key, prevKey: self.prevKey)
         }
 
-        guard let connection = conn else {
-            // Run the callback with a fake failed connection.
-            // Yeah, that means we have an error return value but we still run the callback.
-            let failed = FailedSQLiteDBConnection()
-            let queue = self.sharedConnectionQueue
-            let noConnection = NSError(domain: "mozilla", code: 0, userInfo: [NSLocalizedDescriptionKey: "Could not create a connection"])
-
-            if synchronous {
-                var error: NSError? = nil
-                dispatch_sync(queue) {
-                    error = cb(db: failed)
-                }
-                return error ?? noConnection
-            }
-
-            dispatch_async(queue) {
-                cb(db: failed)
-            }
-
-            return noConnection
-        }
+        // Initialize in case we encounter a nil connection when we go to run the dispatch blocks
+        let failed = FailedSQLiteDBConnection()
```

Comment:
Can we not allocate a new one of these on each successful query? Perhaps move this back down inside the guard?